plugins/lookup/kustomize.py

```diff
@@ -0,0 +1,139 @@
+
+- name: Create kubernetes resources for lookup output
+  k8s:
+    definition: "{{ item }}"
+  with_items:
+    "{{ lookup('kubernetes.core.kustomize', dir='/path/to/kustomization') }}"
+"""
+
+RETURN = """
+  _list:
+    description:
+      - One ore more object definitions returned from the tool execution.
+    type: complex
+    contains:
+      api_version:
+        description: The versioned schema of this representation of an object.
+        returned: success
+        type: str
+      kind:
+        description: Represents the REST resource this object represents.
+        returned: success
+        type: str
+      metadata:
+        description: Standard object metadata. Includes name, namespace, annotations, labels, etc.
+        returned: success
+        type: complex
+      spec:
+        description: Specific attributes of the object. Will vary based on the I(api_version) and I(kind).
+        returned: success
+        type: complex
+      status:
+        description: Current status details for the object.
+        returned: success
+        type: complex
+"""
+
+from ansible.errors import AnsibleLookupError
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils.common.process import get_bin_path
+
+
+import os
+import subprocess
+
+
+def get_binary_from_path(name, opt_dirs=None):
+    opt_arg = {}
+    try:
+        if opt_dirs is not None:
+            if not isinstance(opt_dirs, list):
+                opt_dirs = [opt_dirs]
+            opt_arg['opt_dirs'] = opt_dirs
+        bin_path = get_bin_path(name, **opt_arg)
+        return bin_path
+    except ValueError:
+        return None
+
+
+def run_command(command):
+    cmd = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    return cmd.communicate()
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables=None, dir=".", binary_path=None, opt_dirs=None, **kwargs):
+        executable_path = binary_path
+        if executable_path is None:
+            executable_path = get_binary_from_path(name="kustomize", opt_dirs=opt_dirs)
+            if executable_path is None:
+                executable_path = get_binary_from_path(name="kubectl", opt_dirs=opt_dirs)
+
+            # validate that at least one tool was found
+            if executable_path is None:
+                raise AnsibleLookupError("Failed to find required executable 'kubectl' and 'kustomize' in paths")
+
+        # check input directory
+        kustomization_dir = dir
+        if not os.path.isdir(kustomization_dir):
+            raise AnsibleLookupError("dir parameter {0} is not a valid directory.".format(kustomization_dir))
+        kustomization_file = os.path.join(kustomization_dir, "kustomization.yaml")
+        if not os.path.isfile(kustomization_file):
+            raise AnsibleLookupError("missing 'kustomization.yaml' file from input directory '{0}'".format(kustomization_dir))
```

Comment:
All of this should be removed. We are just duplicating checks that kustomize is already making. This code block is also more restrictive than kustomize. The kustomization file can additionally be named `kustomization.yml` and `Kustomization`.